gradio/pipelines_helpers.py

```diff
@@ -0,0 +1,541 @@
+"""
+Defines internal helper methods for handling transformers and diffusers pipelines.
+These are used by load_from_pipeline method in pipelines.py.
+"""
+
+from typing import Any
+
+from gradio import components
+
+
+def _handle_transformers_pipeline(pipeline: Any) -> dict:
+    try:
+        import transformers
+        from transformers import pipelines
+
+    except ImportError as ie:
+        raise ImportError(
+            "transformers not installed. Please try `pip install transformers`"
+        ) from ie
+
+    # Handle the different pipelines. The has_attr() checks to make sure the pipeline exists in the
+    # version of the transformers library that the user has installed.
+    if hasattr(transformers, "AudioClassificationPipeline") and isinstance(
+        pipeline, pipelines.audio_classification.AudioClassificationPipeline
+    ):
```

Comment:
This repeated segment could be refactored into a helper (repeated nearly similarly for `diffusers`):

```
def is_transformers_pipeline_type(pipeline, class_name: str):
    cls = getattr(transformers, class_name, None)
    return cls and isinstance(pipeline, cls)
```

Reply:
refactored this in the latest commit, thanks for this suggestion.